src/update-feeds/update-feeds.ts

```diff
@@ -56,104 +56,133 @@ export const startUpdateFeedsLoops = async () => {
-        return {
-          firstBatch,
-          dapisCount,
-        };
-      },
-      { totalTimeoutMs: dataFeedUpdateIntervalMs }
-    );
-    if (!goFirstBatch.success) {
-      logger.error(`Failed to get first active dAPIs batch`, goFirstBatch.error);
-      return;
-    }
-    if (Date.now() >= firstBatchStartTime + dataFeedUpdateIntervalMs) {
-      logger.warn(`Fetching the first batch took the whole interval. Skipping updates.`);
-      return;
-    }
+      logger.debug(`Fetching first batch of dAPIs batches`);
+      const firstBatchStartTime = Date.now();
+      const goFirstBatch = await go(
+        async () => {
+          const dapisCountCalldata = dapiDataRegistry.interface.encodeFunctionData('dapisCount');
+          const readDapiWithIndexCalldatas = range(0, dataFeedBatchSize).map((dapiIndex) =>
+            dapiDataRegistry.interface.encodeFunctionData('readDapiWithIndex', [dapiIndex])
+          );
+          const [dapisCountReturndata, ...readDapiWithIndexCallsReturndata] = verifyMulticallResponse(
+            await dapiDataRegistry.callStatic.tryMulticall([dapisCountCalldata, ...readDapiWithIndexCalldatas])
+          );
 
-    const { firstBatch, dapisCount } = goFirstBatch.data;
-    if (dapisCount === 0) {
-      logger.warn(`No active dAPIs found`);
-      return;
-    }
-    const processFirstBatchPromise = processBatch(firstBatch, providerName, provider, chainId);
+          const dapisCount = decodeDapisCountResponse(dapiDataRegistry, dapisCountReturndata!);
+          const firstBatch = readDapiWithIndexCallsReturndata
+            // Because the dapisCount is not known during the multicall, we may ask for non-existent dAPIs. These should be filtered out.
+            .slice(0, dapisCount)
+            .map((dapiReturndata) => ({
+              ...decodeReadDapiWithIndexResponse(dapiDataRegistry, dapiReturndata),
+              chainId,
+            }));
+          return {
+            firstBatch,
+            dapisCount,
+          };
+        },
+        { totalTimeoutMs: dataFeedUpdateIntervalMs }
+      );
+      if (!goFirstBatch.success) {
+        logger.error(`Failed to get first active dAPIs batch`, goFirstBatch.error);
+        return;
+      }
+      if (Date.now() >= firstBatchStartTime + dataFeedUpdateIntervalMs) {
+        logger.warn(`Fetching the first batch took the whole interval. Skipping updates.`);
+        return;
+      }
 
-    // Calculate the stagger time between the rest of the batches.
-    const batchesCount = Math.ceil(dapisCount / dataFeedBatchSize);
-    const staggerTime = batchesCount <= 1 ? 0 : (dataFeedUpdateInterval / batchesCount) * 1000;
+      const { firstBatch, dapisCount } = goFirstBatch.data;
+      if (dapisCount === 0) {
+        logger.warn(`No active dAPIs found`);
+        return;
+      }
+      // NOTE: We need to explicitly handle the .catch here because it's possible that the promise settles before it's
```

Comment:
This gave me a hard time to debug, but its actually correct. There is a `sleep` call below. If the promise resolves sooner it is unhandled (because we await it only a few lines below). To address this issue we need to catch the error ourselves.